File: selenium/test/java/com/thoughtworks/selenium/corebased/TestFramesOpen.java
package com.thoughtworks.selenium.corebased;

import com.thoughtworks.selenium.*;
import org.testng.annotations.*;
import static org.testng.Assert.*;
import java.util.regex.Pattern;

public class TestFramesOpen extends SeleneseTestNgHelper {
	@Test public void testFramesOpen() throws Exception {
		selenium.open("../tests/html/Frames.html");
		selenium.selectFrame("mainFrame");
		verifyTrue(selenium.getLocation().matches("^[\\s\\S]*/tests/html/test_open\\.html$"));
		verifyTrue(selenium.isTextPresent("This is a test of the open command."));
		selenium.open("../tests/html/test_page.slow.html");
		verifyTrue(selenium.getLocation().matches("^[\\s\\S]*/tests/html/test_page\\.slow\\.html$"));
		verifyEquals(selenium.getTitle(), "Slow Loading Page");
		selenium.setTimeout("5000");
		selenium.open("../tests/html/test_open.html");
		selenium.open("../tests/html/test_open.html");
		selenium.open("../tests/html/test_open.html");
	}
}

File: dotnet/test/webdriverbackedselenium/TestFramesOpen.cs
using NUnit.Framework;
using System.Text.RegularExpressions;

namespace Selenium.Tests
{
    [TestFixture]
    public class TestFramesOpen : SeleniumTestCaseBase
    {
        [Test]
        public void ShouldBeAbleToOpenFrames()
        {
            selenium.Open("../tests/html/Frames.html");
            selenium.SelectFrame("mainFrame");
            Assert.IsTrue(Regex.IsMatch(selenium.GetLocation(), "^[\\s\\S]*/tests/html/test_open\\.html$"));
            Assert.IsTrue(selenium.IsTextPresent("This is a test of the open command."));
            selenium.Open("../tests/html/test_page.slow.html");
            Assert.IsTrue(Regex.IsMatch(selenium.GetLocation(), "^[\\s\\S]*/tests/html/test_page\\.slow\\.html$"));
            Assert.AreEqual("Slow Loading Page", selenium.GetTitle());
            selenium.SetTimeout("5000");
            selenium.Open("../tests/html/test_open.html");
            selenium.Open("../tests/html/test_open.html");
            selenium.Open("../tests/html/test_open.html");
        }
    }
}
